---
title: "Polyphosphate kinase (ppk) Gene Presence in Milled vs Unmilled Woodchips"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Files needed:

- "Woodchip_readVrefs_ppk.csv" (output from DIAMOND - copied readVrefs file into csv)
- "Woodchip_metadata_ppk.csv" (total read counts for each sample; read from KBase)

## Load packages & set directory
```{r, include=FALSE}
library(tidyverse)

setwd("C:/github/WBR-microbiome")
```

## Making the plot

### Read in the DIAMOND output data
```{r}
dmnd_out_ppk = read_csv("C:/github/WBR-microbiome/data/Woodchip_readVrefs_ppk.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend,", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_ppk = read_csv("C:/github/WBR-microbiome/data/Woodchip_metadata_ppk.csv") %>%
  rename(fn = filename)
```

### Join the two dataframes together
```{r}
dmnd_ppk = dmnd_out_ppk %>%
    group_by(fn) %>%
    summarize(raw.count = n()) %>%
    left_join(read_count_ppk) %>%
    mutate(norm.count = (raw.count / count) * 1000000) %>% 
    mutate(gene = "ppk") # Add column with gene name (to make plot look the same as N and CAZy plots)
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_ppk = dplyr::group_by(dmnd_ppk, gene, Treatment)

d3_ppk_avg = dplyr::summarize(d3_ppk, mean = mean(norm.count), sd = sd(norm.count))
```

### Create the dot chart
```{r}
p = ggplot(d3_ppk_avg, aes(mean, Treatment)) + geom_point()

p = p +  geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) + expand_limits(x = 0)

p = p + facet_grid(vars(gene))

p = p + theme(panel.background = element_rect(fill = "white"), panel.grid.major.x = element_blank(), strip.background =element_rect(fill="gold"), panel.grid.major.y = element_line(linetype=3, color="dark grey"), legend.position="none", axis.text.y = element_text(size = 10)) #add legend position none.

p = p + labs(x = "Reads per million total reads", y = "Treatment")
  
p
```

### Find dimensions of plot in inches
```{r}
dev.size("in")
```

### Plot with appropriate dimensions
```{r}
tiff("C:/github/WBR-microbiome/figures/ppk_presence_dotchart.tiff", width = 8.4, height = 3, units = 'in', res = 150)
plot(p) # Make plot
dev.off()
```


## Stats
```{r}
rpm_ppk <- d3_ppk

# subset by treatment (U vs M)
rpm_ppk_W <- readspermil %>% 
  filter(Treatment == "U")

rpm_ppk_G <- readspermil %>% 
  filter(Treatment == "M")
```

#### boxplot
```{r}
summary(rpm_ppk_W)
summary(rpm_ppk_G)

#install.packages("ggpubr")
library(ggpubr)

ggplot(rpm_ppk, aes(Treatment, norm.count)) +
  geom_boxplot()
```

#### check assumptions
```{r}
#install.packages("rstatix")
library(rstatix)

# identify outliers
rpm_ppk %>% identify_outliers(norm.count)
### no extreme outliers detected

# check normality
rpm_ppk %>% shapiro_test(norm.count) 
### the two p-values are greater than the significance level 0.05 indicating that 
### the distribution of the data are not significantly different from the normal distribution. 
### In other words, we can assume the normality.

# check normality with Q-Q plots
ggqqplot(rpm_ppk, "norm.count")
### All the points fall approximately along the (45-degree) reference line, for each group. So we can assume normality of the data.
```

#### t-test for read count per million total reads:
```{r}
t.test(norm.count ~ Treatment, data = rpm_ppk, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0089
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.
```


