---
title: "Carbohydrate-Active enZyme (CAZy) Presence in Milled vs Unmilled Woodchips"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Files needed:

- "readVrefs_CAZy.m8" (output from DIAMOND - too large to copy into a csv)
- "Woodchip_metadata_CAZy.csv" (total read counts for each sample; read from KBase)

## Load packages & set directory
```{r, include=FALSE}
library(tidyverse)
```

## Making the plot

### Read in the DIAMOND output data
```{r}
dmnd_out_CAZy <- read_delim(file = "C:/Users/Katie/Documents/WBR-microbiome/data/readVrefs_CAZy.m8")

dmnd_out_CAZy <- dmnd_out_CAZy %>% 
  mutate(source = gsub("\\|.*", "", subject),
         temp = gsub("^[^\\|]+\\||-.*", "", subject),
         classI = gsub("\\|.*", "", temp),
         classII = sub(".*\\|", "", temp))
```

### Correct the sample IDs
```{r}
dmnd_out_CAZy <- dmnd_out_CAZy %>% 
  mutate(fn = ifelse(fn == "./7G.out.daa", "sample7G", fn)) %>% 
  mutate(fn = ifelse(fn == "./8W.out.daa", "sample8W", fn)) %>%
  mutate(fn = ifelse(fn == "./9G.out.daa", "sample9G", fn)) %>%
  mutate(fn = ifelse(fn == "./10W.out.daa", "sample10W", fn)) %>%
  mutate(fn = ifelse(fn == "./11G.out.daa", "sample11G", fn)) %>%
  mutate(fn = ifelse(fn == "./12W.out.daa", "sample12W", fn))
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_CAZy <- read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_metadata_CAZy.csv", col_types = cols(filename = col_character(), Treatment = col_character(), count = col_double())) %>%
  dplyr::rename(fn = filename)
```

### Join the two dataframes together
```{r}
dmnd_CAZy <- dmnd_out_CAZy %>%
  left_join(read_count_CAZy) %>%
  select(query, fn, count, classI)
```

### Gives a count of the AA types
```{r}
dmnd_AAcount <- dmnd_CAZy %>%
  count(classI) %>% 
  filter(classI == "AA0" | classI == "AA1" | classI == "AA1_1" | classI == "AA1_2" | classI == "AA1_3" | classI == "AA2" | classI == "AA3" | classI == "AA3_1" | classI == "AA3_2" | classI == "AA3_3" | classI == "AA4" | classI == "AA5" | classI == "AA5_1" | classI == "AA5_2" | classI == "AA6" | classI == "AA7" | classI == "AA8" | classI == "AA9" | classI == "AA10" | classI == "AA11" | classI == "AA12" | classI == "AA13" | classI == "AA14" | classI == "AA15" | classI == "AA16" | classI == "AA17")
```

### Filter based on AA
```{r}
dmnd_AAfilter <- dmnd_CAZy %>% 
  filter(classI == "AA0" | classI == "AA1" | classI == "AA1_1" | classI == "AA1_2" | classI == "AA1_3" | classI == "AA2" | classI == "AA3" | classI == "AA3_1" | classI == "AA3_2" | classI == "AA3_3" | classI == "AA4" | classI == "AA5" | classI == "AA5_1" | classI == "AA5_2" | classI == "AA6" | classI == "AA7" | classI == "AA8" | classI == "AA9" | classI == "AA10" | classI == "AA11" | classI == "AA12" | classI == "AA13" | classI == "AA14" | classI == "AA15" | classI == "AA16" | classI == "AA17")
```

### Summarize by sample ID
```{r}
d2_CAZy <- dmnd_AAfilter %>%
    group_by(fn, classI) %>%
    summarize(raw.count = n()) %>% #gives number of queries in each sample were mapped to the given gene
    left_join(read_count_CAZy) %>%
    mutate(norm.count = (raw.count / count) * 1000000) #gives reads per million total reads
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_CAZy = dplyr::group_by(d2_CAZy, classI, Treatment)

d3_CAZy_avg = dplyr::summarize(d3_CAZy, mean = mean(norm.count), sd = sd(norm.count))
```

### Add a new column with corrected names
```{r}
d3_CAZy_avg$rename = c("AA0*", "AA0*", "AA1*", "AA1*", "AA10*", "AA10*", "AA11", "AA11", "AA12*", "AA12*", "AA13", "AA13", "AA14*", "AA14*", "AA15", "AA15", "AA16", "AA16", "AA17", "AA17", "AA1_1", "AA1_1", "AA1.2*", "AA1.2*", "AA1.3*", "AA1.3*", "AA2*", "AA2*", "AA3*", "AA3*", "AA3.1*", "AA3.1*", "AA3.2*", "AA3.2*", "AA3.3*", "AA3.3*", "AA4*", "AA4*", "AA5*", "AA5*", "AA5.1*", "AA5.1*", "AA5.2*", "AA5.2*", "AA6", "AA6", "AA7*", "AA7*", "AA8", "AA8", "AA9*", "AA9*")

#adds a column of these values, in this order (FYI - use d3_CAZy_avg to help make this list in the correct order)
```

### Put CAZy AA families in order for the facet
```{r}
d3_CAZy_avg$ord = factor(d3_CAZy_avg$rename, levels=c("AA0*", "AA1*", "AA1.1", "AA1.2*", "AA1.3*", "AA2*", "AA3*", "AA3.1*", "AA3.2*", "AA3.3*", "AA4*", "AA5*", "AA5.1*", "AA5.2*", "AA6", "AA7*", "AA8", "AA9*", "AA10*", "AA11", "AA12*", "AA13", "AA14*", "AA15", "AA16", "AA17"))
```

### Create the dot chart
```{r}
#p = ggplot(d3_CAZy_avg, aes(mean, Treatment)) + geom_point()

#p = p +  geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5))

#p = p + facet_grid(ord ~ .)

#p = p + theme(panel.background = element_rect(fill = "white"), panel.grid.major.x = element_blank(), strip.background =element_rect(fill="gold"), panel.grid.major.y = element_line(linetype=3, color="dark grey"), legend.position="none", axis.text.y = element_text(size = 5)) #add legend position none.

#p = p + labs(x = "Reads per million total reads", y = "Treatment")
  
#p
```


### Trim/subset the plot to exclude AA families with low gene presence, and create the dot chart
```{r}
d3_CAZy_avg_over10 <- d3_CAZy_avg %>% 
  filter(mean > 10)

p2 = ggplot(d3_CAZy_avg_over10, aes(mean, Treatment)) + geom_point()

p2 = p2 +  geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) + expand_limits(x = 0)

p2 = p2 + facet_grid(ord ~ .)

p2 = p2 + theme(panel.background = element_rect(fill = "white"), panel.grid.major.x = element_blank(), strip.background = element_rect(fill="gold"), panel.grid.major.y = element_line(linetype=3, color="dark grey"), legend.position="none", axis.text.y = element_text(size = 10)) #add legend position none.

p2 = p2 + labs(x = "Reads per million total reads", y = "Treatment")
  
p2
```


## Stats
```{r}
rpm_CAZy <- d3_CAZy

# subset by each AA family
rpm_CAZy_AA0 <- rpm_CAZy %>% 
  filter(classI == "AA0")

rpm_CAZy_AA1 <- rpm_CAZy %>% 
  filter(classI == "AA1")

rpm_CAZy_AA1_1 <- rpm_CAZy %>% 
  filter(classI == "AA1_1")

rpm_CAZy_AA1_2 <- rpm_CAZy %>% 
  filter(classI == "AA1_2")

rpm_CAZy_AA1_3 <- rpm_CAZy %>% 
  filter(classI == "AA1_3")

rpm_CAZy_AA2 <- rpm_CAZy %>% 
  filter(classI == "AA2")

rpm_CAZy_AA3 <- rpm_CAZy %>% 
  filter(classI == "AA3")

rpm_CAZy_AA3_1 <- rpm_CAZy %>% 
  filter(classI == "AA3_1")

rpm_CAZy_AA3_2 <- rpm_CAZy %>% 
  filter(classI == "AA3_2")

rpm_CAZy_AA3_3 <- rpm_CAZy %>% 
  filter(classI == "AA3_3")

rpm_CAZy_AA4 <- rpm_CAZy %>% 
  filter(classI == "AA4")

rpm_CAZy_AA5 <- rpm_CAZy %>% 
  filter(classI == "AA5")

rpm_CAZy_AA5_1 <- rpm_CAZy %>% 
  filter(classI == "AA5_1")

rpm_CAZy_AA5_2 <- rpm_CAZy %>% 
  filter(classI == "AA5_2")

rpm_CAZy_AA6 <- rpm_CAZy %>% 
  filter(classI == "AA6")

rpm_CAZy_AA7 <- rpm_CAZy %>% 
  filter(classI == "AA7")

rpm_CAZy_AA8 <- rpm_CAZy %>% 
  filter(classI == "AA8")

rpm_CAZy_AA9 <- rpm_CAZy %>% 
  filter(classI == "AA9")

rpm_CAZy_AA10 <- rpm_CAZy %>% 
  filter(classI == "AA10")

rpm_CAZy_AA11 <- rpm_CAZy %>% 
  filter(classI == "AA11")

rpm_CAZy_AA12 <- rpm_CAZy %>% 
  filter(classI == "AA12")

rpm_CAZy_AA13 <- rpm_CAZy %>% 
  filter(classI == "AA13")

rpm_CAZy_AA14 <- rpm_CAZy %>% 
  filter(classI == "AA14")

rpm_CAZy_AA15 <- rpm_CAZy %>% 
  filter(classI == "AA15")

rpm_CAZy_AA16 <- rpm_CAZy %>% 
  filter(classI == "AA16")

rpm_CAZy_AA17 <- rpm_CAZy %>% 
  filter(classI == "AA17")
```

#### boxplot
```{r}
#install.packages("ggpubr")
library(ggpubr)

# example:
ggplot(rpm_CAZy_AA17, aes(Treatment, norm.count)) +
  geom_boxplot()
```

#### check assumptions
```{r}
#install.packages("rstatix")
library(rstatix)

# identify outliers
rpm_CAZy_AA0 %>% identify_outliers(norm.count)
rpm_CAZy_AA1 %>% identify_outliers(norm.count)
rpm_CAZy_AA1_1 %>% identify_outliers(norm.count)
rpm_CAZy_AA1_2 %>% identify_outliers(norm.count)
rpm_CAZy_AA1_3 %>% identify_outliers(norm.count)
rpm_CAZy_AA2 %>% identify_outliers(norm.count)
rpm_CAZy_AA3 %>% identify_outliers(norm.count)
rpm_CAZy_AA3_1 %>% identify_outliers(norm.count)
rpm_CAZy_AA3_2 %>% identify_outliers(norm.count)
rpm_CAZy_AA3_3 %>% identify_outliers(norm.count)
rpm_CAZy_AA4 %>% identify_outliers(norm.count)
rpm_CAZy_AA5 %>% identify_outliers(norm.count)
rpm_CAZy_AA5_1 %>% identify_outliers(norm.count)
rpm_CAZy_AA5_2 %>% identify_outliers(norm.count)
rpm_CAZy_AA6 %>% identify_outliers(norm.count)
rpm_CAZy_AA7 %>% identify_outliers(norm.count)
rpm_CAZy_AA8 %>% identify_outliers(norm.count)
rpm_CAZy_AA9 %>% identify_outliers(norm.count)
rpm_CAZy_AA10 %>% identify_outliers(norm.count)
rpm_CAZy_AA11 %>% identify_outliers(norm.count)
rpm_CAZy_AA12 %>% identify_outliers(norm.count)
rpm_CAZy_AA13 %>% identify_outliers(norm.count)
rpm_CAZy_AA14 %>% identify_outliers(norm.count)
rpm_CAZy_AA15 %>% identify_outliers(norm.count)
rpm_CAZy_AA16 %>% identify_outliers(norm.count)
rpm_CAZy_AA17 %>% identify_outliers(norm.count)
### no extreme outliers detected

# check normality
rpm_CAZy_AA0 %>% shapiro_test(norm.count) # p-value slightly under with this one, check with Q-Q
rpm_CAZy_AA1 %>% shapiro_test(norm.count)
rpm_CAZy_AA1_1 %>% shapiro_test(norm.count)
rpm_CAZy_AA1_2 %>% shapiro_test(norm.count) # p-value is under with this one, check with Q-Q
rpm_CAZy_AA1_3 %>% shapiro_test(norm.count) 
rpm_CAZy_AA2 %>% shapiro_test(norm.count) # p-value slightly under with this one, check with Q-Q
rpm_CAZy_AA3 %>% shapiro_test(norm.count)
rpm_CAZy_AA3_1 %>% shapiro_test(norm.count)
rpm_CAZy_AA3_2 %>% shapiro_test(norm.count)
rpm_CAZy_AA3_3 %>% shapiro_test(norm.count)
rpm_CAZy_AA4 %>% shapiro_test(norm.count)
rpm_CAZy_AA5 %>% shapiro_test(norm.count)
rpm_CAZy_AA5_1 %>% shapiro_test(norm.count) # p-value is under with this one, check with Q-Q
rpm_CAZy_AA5_2 %>% shapiro_test(norm.count)
rpm_CAZy_AA6 %>% shapiro_test(norm.count)
rpm_CAZy_AA7 %>% shapiro_test(norm.count)
rpm_CAZy_AA8 %>% shapiro_test(norm.count)
rpm_CAZy_AA9 %>% shapiro_test(norm.count) # p-value slightly under with this one, check with Q-Q
rpm_CAZy_AA10 %>% shapiro_test(norm.count)
rpm_CAZy_AA11 %>% shapiro_test(norm.count) # p-value is under with this one, check with Q-Q
rpm_CAZy_AA12 %>% shapiro_test(norm.count)
rpm_CAZy_AA13 %>% shapiro_test(norm.count)
rpm_CAZy_AA14 %>% shapiro_test(norm.count)
rpm_CAZy_AA15 %>% shapiro_test(norm.count)
rpm_CAZy_AA16 %>% shapiro_test(norm.count)
rpm_CAZy_AA17 %>% shapiro_test(norm.count)
### the two p-values are greater than the significance level 0.05 indicating that 
### the distribution of the data are not significantly different from the normal distribution. 
### In other words, we can assume the normality.

# check normality with Q-Q plots
ggqqplot(rpm_CAZy_AA0, "norm.count")
ggqqplot(rpm_CAZy_AA1_2, "norm.count")
ggqqplot(rpm_CAZy_AA2, "norm.count")
ggqqplot(rpm_CAZy_AA5_1, "norm.count")
ggqqplot(rpm_CAZy_AA9, "norm.count")
ggqqplot(rpm_CAZy_AA11, "norm.count")
### All the points fall approximately along the (45-degree) reference line, for each group. So we can assume normality of the data.
```

#### t-test for read count per million total reads:
```{r}
t.test(norm.count ~ Treatment, data = rpm_CAZy_AA0, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0318
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA1, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0391
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA1_1, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.186
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA1_2, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0407
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA1_3, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0523
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA2, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0018
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA3, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0204
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA3_1, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0522
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA3_2, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0157
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA3_3, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0509
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA4, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0300
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA5, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0443
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA5_1, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0381
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA5_2, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0211
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA6, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.1672
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA7, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0423
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA8, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.061
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA9, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0462
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA10, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0477
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA11, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0736
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA12, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0475
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA13, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0600
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA14, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0237
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA15, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.1039
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA16, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0673
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_CAZy_AA17, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.1316
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.
```





