---
title: "Polyphosphate kinase (ppk) Gene Presence in Milled vs Unmilled Woodchips"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Files needed:

- "Woodchip_readVrefs_ppk.csv" (output from DIAMOND - copied readVrefs file into csv)
- "Woodchip_metadata_ppk.csv" (total read counts for each sample; read from KBase)

## Load packages & set directory
```{r, include=FALSE}
#install.packages("tidyverse")
library(tidyverse)
library(ggpubr)
library(rstatix)
```

## Making the plot

### Read in the DIAMOND output data
```{r}
dmnd_out_ppk = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_readVrefs_ppk.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend,", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_ppk = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_metadata_ppk.csv")
```

### Join the two dataframes together
```{r}
dmnd_ppk = dmnd_out_ppk %>%
    group_by(fn) %>%
    summarize(raw.count = n()) %>%
    left_join(read_count_ppk) %>%
    mutate(norm.count = (raw.count / count) * 1000000) %>% 
    mutate(gene = "ppk*") # Add column with gene name (to make plot look the same as N and CAZy plots)
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_ppk = dplyr::group_by(dmnd_ppk, gene, Treatment)

#d3_ppk_avg = dplyr::summarize(d3_ppk, mean = mean(norm.count), sd = sd(norm.count))

d3_ppk_avg = dplyr::summarize(d3_ppk, median = median(norm.count), upper_lim = max(norm.count), lower_lim = min(norm.count))
```

### Create the dot chart
```{r}
d3_ppk_avg %>% 
  ggplot(aes(median, Treatment, color = Treatment)) +
  geom_point() +
  #geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) +
  geom_errorbarh(aes(xmax = upper_lim, xmin = lower_lim, height = .5)) +
  #facet_grid(ord ~ .) +
  scale_color_manual(values = c("darkorange", "deepskyblue2")) +
  theme(panel.background = element_rect(fill = "white"), 
        panel.grid.major.x = element_blank(), 
        strip.background = element_rect(fill = "grey"), 
        panel.grid.major.y = element_line(linetype=3, color = "dark grey"), 
        legend.position ="none", 
        axis.text.y = element_text(size = 10)) +
  labs(x = "Reads per million total reads", y = "Treatment")
```


## Stats

#### check assumptions
```{r}
## using: https://www.datanovia.com/en/lessons/t-test-in-r/

rpm_ppk <- dmnd_ppk

# summary stats
rpm_ppk %>%
  group_by(Treatment) %>%
  get_summary_stats(norm.count, type = "mean_sd")

# summary boxplot
ggboxplot(rpm_ppk, x = "Treatment", y = "norm.count", 
  ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")

# identify outliers
rpm_ppk %>% 
  group_by(Treatment) %>% 
  identify_outliers(norm.count) 
### The result of this function yields zero rows meaning that there were not outliers in our database.

# check normality
## The normality assumption can be checked by computing the Shapiro-Wilk test. 
## If the data is normally distributed, the p-value should be greater than 0.05.
rpm_ppk %>% 
  group_by(Treatment) %>% 
  shapiro_test(norm.count)
### the two p-values are greater than the significance level 0.05 indicating that 
### the distribution of the data are not significantly different from the normal distribution. 
### In other words, we can assume the normality.

ggqqplot(rpm_ppk, x = "norm.count", facet.by = "Treatment")
### All the points fall approximately along the (45-degree) reference line, for each group. So we can assume normality of the data.

# check equality of variances
## This can be done using the Leveneâ€™s test. If the variances of groups are equal, the p-value should be greater than 0.05.
rpm_ppk %>% levene_test(norm.count ~ Treatment)
```

#### Welch's t-test for read count per million total reads:
```{r}
t_test(norm.count ~ Treatment, data = rpm_ppk, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0089
### less than significance level of 0.05, so the means of the U and M groups are significantly different from each other.

## Note about paper - adding "var.equal = FALSE" returns the same p-values. 
## Use this argument if the equal variances assumption was not met, or if you want to be extra safe.
#### With small samples sizes of 10 or fewer it's unlikely that normality tests will detect non-normality. Use Mann-Whitney
```

## Mann-Whitney U Test
```{r}
wilcox_test(norm.count ~ Treatment, data = rpm_ppk, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
# p-value = 0.081
```
