---
title: "Denitrifying (N) Gene Presence in Milled vs Unmilled Woodchips"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Files needed:

- "Woodchip_readVrefs_N.csv" (output from DIAMOND - copied readVrefs file into csv)
- "annotation_map.csv" (created by Jim for denitrifying genes)
- "Woodchip_metadata_N.csv" (total read counts for each sample; read from KBase)

## Load packages & set directory
```{r, include=FALSE}
library(tidyverse)

#setwd("C:/github/WBR-microbiome")
```

## Making the plot

### Read in the DIAMOND output data
```{r}
dmnd_out_Ncycle = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_readVrefs_N.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)
```

### Correct the sample IDs
```{r}
dmnd_out_Ncycle_correct <- dmnd_out_Ncycle %>% 
  mutate(fn = ifelse(fn == "./7G.out.daa", "sample7G", fn)) %>% 
  mutate(fn = ifelse(fn == "./8W.out.daa", "sample8W", fn)) %>%
  mutate(fn = ifelse(fn == "./9G.out.daa", "sample9G", fn)) %>%
  mutate(fn = ifelse(fn == "./10W.out.daa", "sample10W", fn)) %>%
  mutate(fn = ifelse(fn == "./11G.out.daa", "sample11G", fn)) %>%
  mutate(fn = ifelse(fn == "./12W.out.daa", "sample12W", fn))
```

### Read in the annotation map for denitrifying genes
```{r}
annotation_map = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/annotation_map.csv", col_types = "cc", col_names = c("gene","subject"))
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_N = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_metadata_N.csv")
```

### Join the three dataframes together
```{r}
dmnd_N = dmnd_out_Ncycle_correct %>%
    left_join(annotation_map, relationship = "many-to-many") %>%
    group_by(fn, gene) %>%
    summarize(raw.count = n()) %>%
    left_join(read_count_N) %>%
    mutate(norm.count = (raw.count / count) * 1000000) #calculate counts
```

### Add gene names
```{r}
d2_N = dmnd_N[dmnd_N$gene %in% c("Nap", "Nar", "NirK", "NirS", "cNorB", "qNorZ", "NosZ"), ] 
#filters out any rows that are not these 7 genes
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_N = dplyr::group_by(d2_N,gene, Treatment)

d3_N_avg = dplyr::summarize(d3_N, mean = mean(norm.count), sd = sd(norm.count))
```

### Add a new column with corrected names
```{r}
d3_N_avg$rename = c("Nap", "Nap", "Nar", "Nar", "NirK", "NirK", "NirS", "NirS", "NosZ", "NosZ", "cNor", "cNor", "qNor", "qNor")
```

### Put genes in order for the facet
```{r}
d3_N_avg$ord = factor(d3_N_avg$rename, levels=c("Nap","Nar","NirK","NirS","cNor","qNor","NosZ"))
```

### Create the dot chart
```{r}
p = ggplot(d3_N_avg, aes(mean, Treatment)) + geom_point()

p = p +  geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) + expand_limits(x = 0)

p = p + facet_grid(ord ~ .)

p = p + theme(panel.background = element_rect(fill = "white"), panel.grid.major.x = element_blank(), strip.background =element_rect(fill="gold"), panel.grid.major.y = element_line(linetype=3, color="dark grey"), legend.position="none", axis.text.y = element_text(size = 10)) #add legend position none.

p = p + labs(x = "Reads per million total reads", y = "Treatment")
  
p
```


## Stats
```{r}
rpm_N <- d3_N

# subset by each N-cycle gene
rpm_N_nap <- rpm_N %>% 
  filter(gene == "Nap")

rpm_N_nar <- rpm_N %>% 
  filter(gene == "Nar")

rpm_N_nirk <- rpm_N %>% 
  filter(gene == "NirK")

rpm_N_nirs <- rpm_N %>% 
  filter(gene == "NirS")

rpm_N_cnor <- rpm_N %>% 
  filter(gene == "cNorB")

rpm_N_qnor <- rpm_N %>% 
  filter(gene == "qNorZ")

rpm_N_nosz <- rpm_N %>% 
  filter(gene == "NosZ")
```

#### boxplot
```{r}
#install.packages("ggpubr")
library(ggpubr)

# example:
ggplot(rpm_N_nar, aes(Treatment, norm.count)) +
  geom_boxplot()
```

#### check assumptions
```{r}
#install.packages("rstatix")
library(rstatix)

# identify outliers
rpm_N_nap %>% identify_outliers(norm.count)
rpm_N_nar %>% identify_outliers(norm.count)
rpm_N_nirk %>% identify_outliers(norm.count)
rpm_N_nirs %>% identify_outliers(norm.count)
rpm_N_cnor %>% identify_outliers(norm.count)
rpm_N_qnor %>% identify_outliers(norm.count)
rpm_N_nosz %>% identify_outliers(norm.count)
### no extreme outliers detected

# check normality
rpm_N_nap %>% shapiro_test(norm.count) # p-value slightly under with this one, check with Q-Q
rpm_N_nar %>% shapiro_test(norm.count) 
rpm_N_nirk %>% shapiro_test(norm.count) 
rpm_N_nirs %>% shapiro_test(norm.count) 
rpm_N_cnor %>% shapiro_test(norm.count) 
rpm_N_qnor %>% shapiro_test(norm.count) 
rpm_N_nosz %>% shapiro_test(norm.count) 
### the two p-values are greater than the significance level 0.05 indicating that 
### the distribution of the data are not significantly different from the normal distribution. 
### In other words, we can assume the normality.

# check normality with Q-Q plots
ggqqplot(rpm_N_nap, "norm.count")
### All the points fall approximately along the (45-degree) reference line, for each group. So we can assume normality of the data.
```

#### t-test for read count per million total reads:
```{r}
t.test(norm.count ~ Treatment, data = rpm_N_nap, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.5798
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_N_nar, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.5885
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_N_nirk, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.0760
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_N_nirs, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.9136
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_N_cnor, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.1055
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_N_qnor, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.2681
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t.test(norm.count ~ Treatment, data = rpm_N_nosz, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.4111
### NOT less than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.
```


## Create a COMBINED ppk and N-cycle dot plot (for paper)

### Read in the ppk DIAMOND output data
```{r}
dmnd_out_ppk = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_readVrefs_ppk.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend,", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_ppk = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_metadata_ppk.csv")
```

### Join the two dataframes together
```{r}
dmnd_ppk = dmnd_out_ppk %>%
    group_by(fn) %>%
    summarize(raw.count = n()) %>%
    left_join(read_count_ppk) %>%
    mutate(norm.count = (raw.count / count) * 1000000) %>% 
    mutate(gene = "ppk*") # Add column with gene name (to make plot look the same as N and CAZy plots)
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_ppk = dplyr::group_by(dmnd_ppk, gene, Treatment)

d3_ppk_avg = dplyr::summarize(d3_ppk, mean = mean(norm.count), sd = sd(norm.count))
```

### Bind the ppk and N-cycle data together
```{r}
d3_ppk_avg
d3_N_avg

d3_N_ppk <- rbind(d3_N_avg, d3_ppk_avg)

d3_N_ppk$rename = c("Nap", "Nap", "Nar", "Nar", "NirK", "NirK", "NirS", "NirS", "NosZ", "NosZ", "cNor", "cNor", "qNor", "qNor", "ppk*", "ppk*")

d3_N_ppk$ord = factor(d3_N_ppk$rename, levels=c("Nap","Nar","NirK","NirS","cNor","qNor","NosZ", "ppk*"))
```

### Create COMBINED dot chart
```{r}
p = ggplot(d3_N_ppk, aes(mean, Treatment)) + geom_point()

p = p +  geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) + expand_limits(x = 0)

p = p + facet_grid(ord ~ .)

p = p + theme(panel.background = element_rect(fill = "white"), panel.grid.major.x = element_blank(), strip.background =element_rect(fill="gold"), panel.grid.major.y = element_line(linetype=3, color="dark grey"), legend.position="none", axis.text.y = element_text(size = 10)) #add legend position none.

p = p + labs(x = "Reads per million total reads", y = "Treatment")
  
p
```
