---
title: "Denitrifying (N) Gene Presence in Milled vs Unmilled Woodchips"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Files needed:

- "Woodchip_readVrefs_N.csv" (output from DIAMOND - copied readVrefs file into csv)
- "annotation_map.csv" (created by Jim for denitrifying genes)
- "Woodchip_metadata_N.csv" (total read counts for each sample; read from KBase)

## Load packages & set directory
```{r, include=FALSE}
library(tidyverse)
library(ggpubr)
library(rstatix)
```

## Making the plot

### Read in the DIAMOND output data
```{r}
dmnd_out_Ncycle = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_readVrefs_N.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)
```

### Correct the sample IDs
```{r}
dmnd_out_Ncycle_correct <- dmnd_out_Ncycle %>% 
  mutate(fn = ifelse(fn == "./7G.out.daa", "sample7G", fn)) %>% 
  mutate(fn = ifelse(fn == "./8W.out.daa", "sample8W", fn)) %>%
  mutate(fn = ifelse(fn == "./9G.out.daa", "sample9G", fn)) %>%
  mutate(fn = ifelse(fn == "./10W.out.daa", "sample10W", fn)) %>%
  mutate(fn = ifelse(fn == "./11G.out.daa", "sample11G", fn)) %>%
  mutate(fn = ifelse(fn == "./12W.out.daa", "sample12W", fn))
```

### Read in the annotation map for denitrifying genes
```{r}
annotation_map = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/annotation_map.csv", col_types = "cc", col_names = c("gene","subject"))
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_N = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_metadata_N.csv")
```

### Join the three dataframes together
```{r}
dmnd_N = dmnd_out_Ncycle_correct %>%
    left_join(annotation_map, relationship = "many-to-many") %>%
    group_by(fn, gene) %>%
    summarize(raw.count = n()) %>%
    left_join(., read_count_N, by = "fn") %>%
    mutate(norm.count = (raw.count / count) * 1000000) #calculate counts
```

### Add gene names
```{r}
d2_N = dmnd_N[dmnd_N$gene %in% c("Nap", "Nar", "NirK", "NirS", "cNorB", "qNorZ", "NosZ"), ] 
#filters out any rows that are not these 7 genes
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_N = dplyr::group_by(d2_N, gene, Treatment)

#d3_N_avg = dplyr::summarize(d3_N, mean = mean(norm.count), sd = sd(norm.count))

d3_N_avg = dplyr::summarize(d3_N, median = median(norm.count), upper_lim = max(norm.count), lower_lim = min(norm.count))
```

### Add a new column with corrected names
```{r}
d3_N_avg$rename = c("Nap", "Nap", "Nar", "Nar", "NirK", "NirK", "NirS", "NirS", "NosZ", "NosZ", "cNor", "cNor", "qNor", "qNor")
```

### Put genes in order for the facet
```{r}
d3_N_avg$ord = factor(d3_N_avg$rename, levels=c("Nap","Nar","NirK","NirS","cNor","qNor","NosZ"))
```

### Create the dot chart
```{r}
d3_N_avg %>% 
  ggplot(aes(median, Treatment, color = Treatment)) +
  geom_point() +
  #geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) +
  geom_errorbarh(aes(xmax = upper_lim, xmin = lower_lim, height = .5)) +
  facet_grid(ord ~ .) +
  scale_color_manual(values = c("darkorange", "deepskyblue2")) +
  theme(panel.background = element_rect(fill = "white"), 
        panel.grid.major.x = element_blank(), 
        strip.background = element_rect(fill = "grey"), 
        panel.grid.major.y = element_line(linetype=3, color = "dark grey"), 
        legend.position ="none", 
        axis.text.y = element_text(size = 10)) +
  labs(x = "Reads per million total reads", y = "Treatment")
```


## Stats
```{r}
rpm_N <- dmnd_N %>% 
  ungroup()

# subset by each N-cycle gene
rpm_N_nap <- rpm_N %>% 
  filter(gene == "Nap")

rpm_N_nar <- rpm_N %>% 
  filter(gene == "Nar")

rpm_N_nirk <- rpm_N %>% 
  filter(gene == "NirK")

rpm_N_nirs <- rpm_N %>% 
  filter(gene == "NirS")

rpm_N_cnor <- rpm_N %>% 
  filter(gene == "cNorB")

rpm_N_qnor <- rpm_N %>% 
  filter(gene == "qNorZ")

rpm_N_nosz <- rpm_N %>% 
  filter(gene == "NosZ")
```

#### check assumptions
```{r}
## using: https://www.datanovia.com/en/lessons/t-test-in-r/

# summary stats
rpm_N_nap %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")
rpm_N_nar %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")
rpm_N_nirk %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")
rpm_N_nirs %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")
rpm_N_cnor %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")
rpm_N_qnor %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")
rpm_N_nosz %>% group_by(Treatment) %>% get_summary_stats(norm.count, type = "mean_sd")

# summary boxplot
ggboxplot(rpm_N_nap, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")
ggboxplot(rpm_N_nar, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")
ggboxplot(rpm_N_nirk, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")
ggboxplot(rpm_N_nirs, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")
ggboxplot(rpm_N_cnor, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")
ggboxplot(rpm_N_qnor, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")
ggboxplot(rpm_N_nosz, x = "Treatment", y = "norm.count", ylab = "Reads per Million Total", xlab = "Treatment", add = "jitter")

# identify outliers
rpm_N_nap %>% group_by(Treatment) %>% identify_outliers(norm.count) 
rpm_N_nar %>% group_by(Treatment) %>% identify_outliers(norm.count) 
rpm_N_nirk %>% group_by(Treatment) %>% identify_outliers(norm.count) 
rpm_N_nirs %>% group_by(Treatment) %>% identify_outliers(norm.count) 
rpm_N_cnor %>% group_by(Treatment) %>% identify_outliers(norm.count) 
rpm_N_qnor %>% group_by(Treatment) %>% identify_outliers(norm.count) 
rpm_N_nosz %>% group_by(Treatment) %>% identify_outliers(norm.count) 
### The result of this function yields zero rows meaning that there were not outliers in our database.

# check normality
## The normality assumption can be checked by computing the Shapiro-Wilk test. 
## If the data is normally distributed, the p-value should be greater than 0.05.
rpm_N_nap %>% group_by(Treatment) %>% shapiro_test(norm.count) #U in this one is 0.03
rpm_N_nar %>% group_by(Treatment) %>% shapiro_test(norm.count)
rpm_N_nirk %>% group_by(Treatment) %>% shapiro_test(norm.count)
rpm_N_nirs %>% group_by(Treatment) %>% shapiro_test(norm.count)
rpm_N_cnor %>% group_by(Treatment) %>% shapiro_test(norm.count)
rpm_N_qnor %>% group_by(Treatment) %>% shapiro_test(norm.count)
rpm_N_nosz %>% group_by(Treatment) %>% shapiro_test(norm.count)

ggqqplot(rpm_N_nap, x = "norm.count", facet.by = "Treatment")
ggqqplot(rpm_N_nar, x = "norm.count", facet.by = "Treatment")
ggqqplot(rpm_N_nirk, x = "norm.count", facet.by = "Treatment")
ggqqplot(rpm_N_nirs, x = "norm.count", facet.by = "Treatment")
ggqqplot(rpm_N_cnor, x = "norm.count", facet.by = "Treatment") 
ggqqplot(rpm_N_qnor, x = "norm.count", facet.by = "Treatment") 
ggqqplot(rpm_N_nosz, x = "norm.count", facet.by = "Treatment")
### Above Q-Q plots: all the points fall approximately along the reference line for each group (so we can assume normality)
### Plus, the two p-values are greater than the significance level 0.05 indicating that 
### the distribution of the data are not significantly different from the normal distribution. 
### In other words, we can assume the normality.

# check equality of variances
## This can be done using the Leveneâ€™s test. If the variances of groups are equal, the p-value should be greater than 0.05.
rpm_N_nap %>% levene_test(norm.count ~ Treatment)
rpm_N_nar %>% levene_test(norm.count ~ Treatment)
rpm_N_nirk %>% levene_test(norm.count ~ Treatment)
rpm_N_nirs %>% levene_test(norm.count ~ Treatment)
rpm_N_cnor %>% levene_test(norm.count ~ Treatment)
rpm_N_qnor %>% levene_test(norm.count ~ Treatment)
rpm_N_nosz %>% levene_test(norm.count ~ Treatment)
```

#### Welch's t-test for read count per million total reads:
```{r}
t_test(norm.count ~ Treatment, data = rpm_N_nap, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.58
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t_test(norm.count ~ Treatment, data = rpm_N_nar, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.589
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t_test(norm.count ~ Treatment, data = rpm_N_nirk, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.076
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t_test(norm.count ~ Treatment, data = rpm_N_nirs, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.914
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t_test(norm.count ~ Treatment, data = rpm_N_cnor, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.105
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t_test(norm.count ~ Treatment, data = rpm_N_qnor, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.268
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

t_test(norm.count ~ Treatment, data = rpm_N_nosz, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
### p-value = 0.411
### greater than significance level of 0.05, so the means of the U and M groups are NOT significantly different from each other.

## Note about paper - adding "var.equal = FALSE" returns the same p-values. 
## Use this argument if the equal variances assumption was not met, or if you want to be extra safe.
#### With small samples sizes of 10 or fewer it's unlikely that normality tests will detect non-normality. Use Mann-Whitney
```

## Mann-Whitney U Test
```{r}
wilcox_test(norm.count ~ Treatment, data = rpm_N_nap, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 0.663

wilcox_test(norm.count ~ Treatment, data = rpm_N_nar, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 1

wilcox_test(norm.count ~ Treatment, data = rpm_N_nirk, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 0.081

wilcox_test(norm.count ~ Treatment, data = rpm_N_nirs, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 1

wilcox_test(norm.count ~ Treatment, data = rpm_N_cnor, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 0.081

wilcox_test(norm.count ~ Treatment, data = rpm_N_qnor, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 0.383

wilcox_test(norm.count ~ Treatment, data = rpm_N_nosz, alternative = "two.sided", paired = FALSE, conf.level = 0.95, exact = FALSE)
### p-value = 0.663
```


## Create a COMBINED ppk and N-cycle dot plot (for paper)

### Read in the ppk DIAMOND output data
```{r}
dmnd_out_ppk = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_readVrefs_ppk.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend,", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)
```

### Read in the metadata (i.e., read counts)
```{r}
read_count_ppk = read_csv("C:/Users/Katie/Documents/WBR-microbiome/data/Woodchip_metadata_ppk.csv")
```

### Join the two dataframes together
```{r}
dmnd_ppk = dmnd_out_ppk %>%
    group_by(fn) %>%
    summarize(raw.count = n()) %>%
    left_join(read_count_ppk) %>%
    mutate(norm.count = (raw.count / count) * 1000000) %>% 
    mutate(gene = "ppk*") # Add column with gene name (to make plot look the same as N and CAZy plots)
```

### Group by treatment (i.e., "G" for milled, "W" for unmilled)
```{r}
d3_ppk = dplyr::group_by(dmnd_ppk, gene, Treatment)

#d3_ppk_avg = dplyr::summarize(d3_ppk, mean = mean(norm.count), sd = sd(norm.count))

d3_ppk_avg = dplyr::summarize(d3_ppk, median = median(norm.count), upper_lim = max(norm.count), lower_lim = min(norm.count))
```

### Bind the ppk and N-cycle data together
```{r}
d3_ppk_avg
d3_N_avg

d3_N_ppk <- rbind(d3_N_avg, d3_ppk_avg)

d3_N_ppk$rename = c("Nap", "Nap", "Nar", "Nar", "NirK", "NirK", "NirS", "NirS", "NosZ", "NosZ", "cNor", "cNor", "qNor", "qNor", "ppk*", "ppk*")

d3_N_ppk$ord = factor(d3_N_ppk$rename, levels=c("Nap","Nar","NirK","NirS","cNor","qNor","NosZ", "ppk*"))
```

### Create COMBINED dot chart
```{r}
d3_N_ppk %>% 
  ggplot(aes(median, Treatment, color = Treatment)) +
  geom_point() +
  #geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) +
  geom_errorbarh(aes(xmax = upper_lim, xmin = lower_lim, height = .5)) +
  facet_grid(ord ~ .) +
  scale_color_manual(values = c("darkorange", "deepskyblue2")) +
  theme(panel.background = element_rect(fill = "white"), 
        panel.grid.major.x = element_blank(), 
        strip.background = element_rect(fill = "grey"), 
        panel.grid.major.y = element_line(linetype=3, color = "dark grey"), 
        legend.position ="none", 
        axis.text.y = element_text(size = 10)) +
  labs(x = "Reads per million total reads", y = "Treatment")
```
