---
title: "Chap 2 - Polyphosphate kinase (ppk) Differential Expression"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Files needed:

- "RNA_readVrefs_ppkTODAY.csv" (output from DIAMOND - copied readVrefs file into csv)
- "readVrefs_CAZy_TODAY.m8" (output from DIAMOND - too large to copy into a csv)
- "RNA_readVrefs_NcycleTODAY.csv" (output from DIAMOND - copied readVrefs file into csv)
- "sampleinfo_RNA.csv" (metadata about each sample)

## Load packages & set directory
```{r, include=FALSE}
#install.packages("tidyverse")
library(tidyverse)
library(pals)
library(ggpubr)
#install.packages("paletteer")
library(paletteer)
library(vegan)
```

## Load data
```{r}
# count data (from DIAMOND)
## ppk
dmnd_out_ppk <- read_csv("data/RNA/chap2_readVrefs_ppk2.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend,", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)

## N cycle
dmnd_out_Ncycle = read_csv("C:/github/WBR-microbiome/data/RNA/chap2_readVrefs_Ncycle.csv", 
                  col_types = "ccdiiiiiiiddc", 
                  col_names = c("query", 
                                "subject", 
                                "pid",
                                "aln_len", 
                                "mismatches", 
                                "gaps", 
                                "qstart", 
                                "qend", 
                                "sstart", 
                                "send", 
                                "e-value", 
                                "bit", 
                                "fn")) %>%
    filter(pid >= 60, aln_len > 25)

## CAZy
dmnd_out_CAZy <- read_delim(file = "data/RNA/readVrefs_CAZy.m8")

colnames(dmnd_out_CAZy) <- c("query", "subject", "pid", "aln_len", "mismatches", "gaps", 
                             "qstart", "qend", "sstart", "send", "e-value", "bit", "fn")

dmnd_out_CAZy <- dmnd_out_CAZy %>% 
  mutate(source = gsub("\\|.*", "", subject),
         temp = gsub("^[^\\|]+\\||-.*", "", subject),
         classI = gsub("\\|.*", "", temp),
         classII = sub(".*\\|", "", temp))

dmnd_out_ppk <- dmnd_out_ppk %>% 
  mutate(fn = ifelse(fn == "./100_trim_cut_181049_201_1.out.daa", "100_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./139_trim_cut_181049_226_1.out.daa", "139_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./140_trim_cut_181049_261_1.out.daa", "140_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./141_trim_cut_181049_219_1.out.daa", "141_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./142_trim_cut_181049_225_1.out.daa", "142_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./147_trim_cut_181049_231_1.out.daa", "147_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./148_trim_cut_181049_220_1.out.daa", "148_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./149_trim_cut_181049_224_1.out.daa", "149_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./150_trim_cut_181049_233_1.out.daa", "150_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./39_trim_cut_181049_183_1.out.daa", "39_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./40_trim_cut_181049_195_1.out.daa", "40_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./41_trim_cut_181049_188_1.out.daa", "41_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./42_trim_cut_181049_194_1.out.daa", "42_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./47_trim_cut_181049_190_1.out.daa", "47_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./48_trim_cut_181049_192_1.out.daa", "48_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./49_trim_cut_181049_186_1.out.daa", "49_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./50_trim_cut_181049_187_1.out.daa", "50_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./89_trim_cut_181049_216_1.out.daa", "89_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./90_trim_cut_181049_214_1.out.daa", "90_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./91_trim_cut_181049_205_1.out.daa", "91_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./92_trim_cut_181049_203_1.out.daa", "92_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./97_trim_cut_181049_210_1.out.daa", "97_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./98_trim_cut_181049_204_1.out.daa", "98_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./99_trim_cut_181049_211_1.out.daa", "99_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./blank_trim_cut_181049_223_1.out.daa", "blank_trim_cut", fn))

dmnd_out_Ncycle <- dmnd_out_Ncycle %>% 
  mutate(fn = ifelse(fn == "./100_trim_cut_181049_201_1.out.daa", "100_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./139_trim_cut_181049_226_1.out.daa", "139_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./140_trim_cut_181049_261_1.out.daa", "140_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./141_trim_cut_181049_219_1.out.daa", "141_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./142_trim_cut_181049_225_1.out.daa", "142_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./147_trim_cut_181049_231_1.out.daa", "147_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./148_trim_cut_181049_220_1.out.daa", "148_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./149_trim_cut_181049_224_1.out.daa", "149_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./150_trim_cut_181049_233_1.out.daa", "150_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./39_trim_cut_181049_183_1.out.daa", "39_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./40_trim_cut_181049_195_1.out.daa", "40_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./41_trim_cut_181049_188_1.out.daa", "41_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./42_trim_cut_181049_194_1.out.daa", "42_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./47_trim_cut_181049_190_1.out.daa", "47_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./48_trim_cut_181049_192_1.out.daa", "48_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./49_trim_cut_181049_186_1.out.daa", "49_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./50_trim_cut_181049_187_1.out.daa", "50_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./89_trim_cut_181049_216_1.out.daa", "89_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./90_trim_cut_181049_214_1.out.daa", "90_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./91_trim_cut_181049_205_1.out.daa", "91_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./92_trim_cut_181049_203_1.out.daa", "92_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./97_trim_cut_181049_210_1.out.daa", "97_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./98_trim_cut_181049_204_1.out.daa", "98_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./99_trim_cut_181049_211_1.out.daa", "99_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./blank_trim_cut_181049_223_1.out.daa", "blank_trim_cut", fn))

dmnd_out_CAZy <- dmnd_out_CAZy %>% 
  mutate(fn = ifelse(fn == "./100_trim_cut_181049_201_1.out.daa", "100_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./139_trim_cut_181049_226_1.out.daa", "139_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./140_trim_cut_181049_261_1.out.daa", "140_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./141_trim_cut_181049_219_1.out.daa", "141_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./142_trim_cut_181049_225_1.out.daa", "142_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./147_trim_cut_181049_231_1.out.daa", "147_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./148_trim_cut_181049_220_1.out.daa", "148_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./149_trim_cut_181049_224_1.out.daa", "149_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./150_trim_cut_181049_233_1.out.daa", "150_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./39_trim_cut_181049_183_1.out.daa", "39_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./40_trim_cut_181049_195_1.out.daa", "40_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./41_trim_cut_181049_188_1.out.daa", "41_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./42_trim_cut_181049_194_1.out.daa", "42_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./47_trim_cut_181049_190_1.out.daa", "47_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./48_trim_cut_181049_192_1.out.daa", "48_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./49_trim_cut_181049_186_1.out.daa", "49_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./50_trim_cut_181049_187_1.out.daa", "50_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./89_trim_cut_181049_216_1.out.daa", "89_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./90_trim_cut_181049_214_1.out.daa", "90_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./91_trim_cut_181049_205_1.out.daa", "91_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./92_trim_cut_181049_203_1.out.daa", "92_trim_cut", fn)) %>% 
  mutate(fn = ifelse(fn == "./97_trim_cut_181049_210_1.out.daa", "97_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./98_trim_cut_181049_204_1.out.daa", "98_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./99_trim_cut_181049_211_1.out.daa", "99_trim_cut", fn)) %>%
  mutate(fn = ifelse(fn == "./blank_trim_cut_181049_223_1.out.daa", "blank_trim_cut", fn))
```

### Read in the metadata (i.e., read counts)
```{r}
read_count = read.csv("data/RNA/chap2_woodchip_metadata.csv")
```

### Read in the annotation map for denitrifying genes
```{r}
annotation_map = read_csv("data/annotation_map.csv", col_types = "cc", col_names = c("gene","subject"))
```

### Join the two dataframes together
```{r}
dmnd_N = dmnd_out_Ncycle %>%
    left_join(annotation_map, join_by(subject), multiple = "all") %>%
    group_by(fn, gene) %>%
    summarize(raw.count = n()) %>% # calculate mapped read count
    left_join(., read_count, by = "fn") %>%
    mutate(norm.count = (raw.count / count) * 1000000) %>% # normalize the mapped read counts
    filter(!fn == "blank_trim_cut")

dmnd_ppk <-  dmnd_out_ppk %>%
    group_by(fn) %>%
    summarize(raw.count = n()) %>% #sums up the rows for each sample
    left_join(read_count) %>%
    mutate(norm.count = (raw.count/count) * 1000000) %>% 
    mutate(gene = "ppk") %>% # Add column with gene name (one gene so annoation map not needed)
    filter(!fn == "blank_trim_cut")

dmnd_CAZy <- dmnd_out_CAZy %>%
  left_join(read_count) %>%
  select(query, fn, count, classI)

dmnd_AAfilter <- dmnd_CAZy %>% 
  filter(classI == "AA0" | classI == "AA1" | classI == "AA1_1" | classI == "AA1_2" | classI == "AA1_3" | classI == "AA2" | classI == "AA3" | classI == "AA3_1" | classI == "AA3_2" | classI == "AA3_3" | classI == "AA4" | classI == "AA5" | classI == "AA5_1" | classI == "AA5_2" | classI == "AA6" | classI == "AA7" | classI == "AA8" | classI == "AA9" | classI == "AA10" | classI == "AA11" | classI == "AA12" | classI == "AA13" | classI == "AA14" | classI == "AA15" | classI == "AA16" | classI == "AA17")

d2_CAZy <- dmnd_AAfilter %>%
    group_by(fn, classI) %>%
    summarize(raw.count = n()) %>% #gives number of queries in each sample were mapped to the given gene
    left_join(read_count) %>%
    mutate(norm.count = (raw.count / count) * 1000000) #gives reads per million total reads
```

### Add gene names for N cycle
```{r}
d2_N = dmnd_N[dmnd_N$gene %in% c("Nap", "Nar", "NirK", "NirS", "cNorB", "qNorZ", "NosZ"), ] 
#filters out any rows that are not these 7 genes
```

### Group by treatment, reactor, and timepoint & calculate plot stats
```{r}
## N cycle
d3_N = dplyr::group_by(d2_N, gene, Treatment, Reactor, Timepoint)

#d3_N_avg = dplyr::summarize(d3_N, mean = mean(norm.count), sd = sd(norm.count))

d3_N_avg = dplyr::summarize(d3_N, median = median(norm.count), upper_lim = max(norm.count), lower_lim = min(norm.count))

## ppk
d3_ppk = dplyr::group_by(dmnd_ppk, gene, Treatment, Reactor, Timepoint)

#d3_ppk_avg = dplyr::summarize(d3_ppk, mean = mean(norm.count), sd = sd(norm.count))

d3_ppk_avg = dplyr::summarize(d3_ppk, median = median(norm.count), upper_lim = max(norm.count), lower_lim = min(norm.count))

## CAZy
d3_CAZy = dplyr::group_by(d2_CAZy, classI, Treatment, Reactor, Timepoint)

#d3_CAZy_avg = dplyr::summarize(d3_CAZy, mean = mean(norm.count), sd = sd(norm.count))

d3_CAZy_avg = dplyr::summarize(d3_CAZy, median = median(norm.count), upper_lim = max(norm.count), lower_lim = min(norm.count))
```

### Add a new column to CAZy set with corrected names
```{r}
d3_CAZy_avg$rename = c("AA0", "AA0", "AA0", "AA0", "AA0", "AA0", "AA0", "AA0", "AA0", "AA0", "AA0", "AA1", "AA1", "AA1", "AA1", "AA1", "AA1", "AA1", "AA1", "AA1", "AA1", "AA1", "AA10", "AA10", "AA10", "AA10", "AA10", "AA10", "AA10", "AA10", "AA10", "AA10", "AA10", "AA11", "AA11", "AA11", "AA11", "AA11", "AA12", "AA12", "AA12", "AA12", "AA12", "AA12", "AA12", "AA12", "AA12", "AA12", "AA14", "AA15", "AA15", "AA15", "AA15", "AA15", "AA15", "AA15", "AA15", "AA15", "AA15", "AA15", "AA16", "AA17", "AA17", "AA17", "AA17", "AA17", "AA17", "AA17", "AA17", "AA17", "AA17", "AA17", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.1", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.2", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA1.3", "AA2", "AA2", "AA2", "AA2", "AA2", "AA2", "AA2", "AA2", "AA2", "AA2", "AA2", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3", "AA3.1", "AA3.1", "AA3.1", "AA3.1", "AA3.1", "AA3.1", "AA3.1", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.2", "AA3.3", "AA3.3", "AA3.3", "AA3.3", "AA3.3", "AA3.3", "AA3.3", "AA3.3", "AA3.3", "AA4", "AA4", "AA4", "AA4", "AA4", "AA4", "AA4", "AA4", "AA4", "AA4", "AA4", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5", "AA5.1", "AA5", "AA5.1", "AA5", "AA5.1", "AA5", "AA5.1", "AA5", "AA5", "AA5.1", "AA5.1", "AA5.2", "AA5.2", "AA5.2", "AA5.2", "AA5.2", "AA5.2", "AA6", "AA6", "AA6", "AA6", "AA6", "AA6", "AA6", "AA6", "AA6", "AA6", "AA6", "AA7", "AA7", "AA7", "AA7", "AA7", "AA7", "AA7", "AA7", "AA8", "AA9", "AA9", "AA9", "AA9", "AA9", "AA9", "AA9", "AA9", "AA9", "AA9", "AA9")

#adds a column of these values, in this order (FYI - use d3_CAZy_avg to help make this list in the correct order)
```

### Bind the ppk and N-cycle data together and rename/reorder genes
```{r}
d3_ppk_avg
d3_N_avg

d3_N_ppk <- rbind(d3_N_avg, d3_ppk_avg)

d3_N_ppk$rename = c("Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nap", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "Nar", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirK", "NirS", "NirS", "NirS", "NirS", "NirS", "NirS", "NirS", "NirS", "NirS", "NirS", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "NosZ", "cNor", "cNor", "cNor", "cNor", "cNor", "cNor", "cNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "qNor", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk", "ppk")

d3_N_ppk$ord = factor(d3_N_ppk$rename, levels=c("Nap","Nar","NirK","NirS","cNor","qNor","NosZ", "ppk"))

d3_CAZy_avg$ord = factor(d3_CAZy_avg$rename, levels=c("AA0", "AA1", "AA1.1", "AA1.2", "AA1.3", "AA2", "AA3", "AA3.1", "AA3.2", "AA3.3", "AA4", "AA5", "AA5.1", "AA5.2", "AA6", "AA7", "AA8", "AA9", "AA10", "AA11", "AA12", "AA13", "AA14", "AA15", "AA16", "AA17"))
```

### Create the dot charts
```{r}
# N cycle and ppk together fct_reorder(taxon_name, percent)
d3_N_ppk$Timepoint <- factor(d3_N_ppk$Timepoint, levels=c('24', '120'))

d3_N_ppk %>% 
  ggplot(aes(median, Reactor, color = Timepoint)) +
  geom_point() +
  #geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) +
  geom_errorbarh(aes(xmax = upper_lim, xmin = lower_lim, height = .5)) +
  facet_grid(ord ~ Treatment) +
  scale_color_manual(values = c("deepskyblue3", "red3")) +
  theme(panel.background = element_rect(fill = "white"), 
        panel.grid.major.x = element_blank(), 
        strip.background = element_rect(fill = "grey"), 
        panel.grid.major.y = element_line(linetype=3, color = "dark grey"), 
        legend.position ="bottom", 
        legend.margin=margin(0,0,0,0),
        axis.text.y = element_text(size = 10)) +
  xlim(c(0,200)) +
  labs(x = "Reads per million total reads", y = "Reactor")

# CAZY
d3_CAZy_avg$Timepoint <- factor(d3_CAZy_avg$Timepoint, levels=c('24', '120'))

d3_CAZy_avg %>% 
  ggplot(aes(median, Reactor, color = Timepoint)) +
  geom_point() +
  #geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) +
  geom_errorbarh(aes(xmax = upper_lim, xmin = lower_lim, height = .5)) +
  facet_grid(ord ~ Treatment) +
  scale_color_manual(values = c("deepskyblue3", "red3")) +
  theme(panel.background = element_rect(fill = "white"), 
        panel.grid.major.x = element_blank(), 
        strip.background = element_rect(fill = "grey"), 
        panel.grid.major.y = element_line(linetype=3, color = "dark grey"), 
        legend.position ="bottom", 
        legend.margin=margin(0,0,0,0),
        axis.text.y = element_text(size = 10)) +
  xlim(c(0,200)) +
  labs(x = "Reads per million total reads", y = "Reactor")

## numbers for paper:
CAZy_N_ppk <- rbind(d3_CAZy_avg, d3_N_ppk) 

CAZy_N_ppk %>% 
  select(c(Treatment, Reactor, Timepoint, upper_lim, rename)) %>% 
  group_by(rename) %>% 
  summarize(max = max(upper_lim)) %>% 
  View()

CAZy_N_ppk %>% 
  filter(Timepoint == "24") %>% 
  View()

CAZy_N_ppk %>% 
  filter(gene == "ppk") %>% 
  View()

CAZy_N_ppk %>% 
  filter(rename == "Nap") %>% 
  View()

CAZy_N_ppk %>% 
  filter(rename == "NirK" | rename == "NirS") %>% 
  View()
```

### Create similar plot for nutrient concentrations
```{r}
icp_ic_results <- read_csv("data/GFBR_icp_ic_results.csv") %>% 
  unite(datetime, c(date, time), sep = " ", remove = FALSE) #combine date and time

RNA_nutrient_conc <- icp_ic_results %>% 
  filter(cycle == 2) %>% 
  filter(sample_timepoint == 24 | sample_timepoint == 120)

conc_plot <- RNA_nutrient_conc %>%
  select(!c(ICP_IC_batch, cycle, hours_after_start, replicate, ICP_sample, 
            IC_sample, RNA_sample, datetime, date, time)) %>% 
  pivot_longer(cols = c("initial_p", "initial_no3", "Al", "Ca", "Fe", 
                        "Mg", "Mn", "P", "S", "Cl", "NO2", "NO3", "PO4", 
                        "SO4"), names_to = "nutrient", values_to = "concentration") %>% 
  group_by(nutrient, treatment, reactor, sample_timepoint) %>% 
  summarize(median = median(concentration), 
            upper_lim = max(concentration), 
            lower_lim = min(concentration))

conc_plot$sample_timepoint <- as.character(conc_plot$sample_timepoint)

conc_plot %>% 
  filter(!nutrient == "initial_p") %>% # turn these off/on depending one what needs to be shown
  filter(!nutrient == "initial_no3") %>%
  filter(!nutrient == "Ca") %>%
  filter(!nutrient == "Cl") %>%
  filter(!nutrient == "Mg") %>% 
  filter(!nutrient == "Fe") %>%
  filter(!nutrient == "NO3") %>%
  filter(!nutrient == "S") %>%
  filter(!nutrient == "SO4") %>%
  ggplot(aes(median, reactor, color = sample_timepoint)) +
  geom_point() +
  #geom_errorbarh(aes(xmax = mean + sd, xmin = mean - sd, height = .5)) +
  geom_errorbarh(aes(xmax = upper_lim, xmin = lower_lim, height = .5)) +
  facet_grid(nutrient ~ treatment) +
  scale_color_manual(values = c("red3", "deepskyblue3")) +
  theme(panel.background = element_rect(fill = "white"), 
        panel.grid.major.x = element_blank(), 
        strip.background = element_rect(fill = "grey"), 
        panel.grid.major.y = element_line(linetype=3, color = "dark grey"), 
        legend.position ="bottom", 
        axis.text.y = element_text(size = 10)) +
  labs(x = "Concentration (mg/L)", y = "Hours post DRW reactor re-flood")
```


## Prep for Kaiju

### Subset the m8 files for read names mapping to the genes
```{r}
# ppk
ppk_hits <- dmnd_out_ppk %>% 
  select(query)

ppk_sub <- ppk_querycol[duplicated(ppk_querycol), ]

## N cycle
# nap
nap <- annotation_map %>% 
  filter(gene == "Nap")

nap_hits <- dmnd_out_Ncycle %>% 
  select(c(query, subject)) %>% 
  filter(subject %in% nap$subject) %>% 
  select(query)

nap_sub <- nap_hits[duplicated(nap_hits), ]

# nar
nar <- annotation_map %>% 
  filter(gene == "Nar")

nar_hits <- dmnd_out_Ncycle %>% 
  select(c(query, subject)) %>% 
  filter(subject %in% nar$subject) %>% 
  select(query)

nar_sub <- nar_hits[duplicated(nar_hits), ]

# nirk
nirk <- annotation_map %>% 
  filter(gene == "NirK")

nirk_hits <- dmnd_out_Ncycle %>% 
  select(c(query, subject)) %>% 
  filter(subject %in% nirk$subject) %>% 
  select(query)

nirk_sub <- nirk_hits[duplicated(nirk_hits), ]

# qnor
qnor <- annotation_map %>% 
  filter(gene == "qNorZ")

qnor_hits <- dmnd_out_Ncycle %>% 
  select(c(query, subject)) %>% 
  filter(subject %in% qnor$subject) %>% 
  select(query)

qnor_sub <- qnor_hits[duplicated(qnor_hits), ]

## CAZy
# AA0
AA0_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA0") %>% 
  select(query)

AA0_sub <- AA0_hits[duplicated(AA0_hits), ]

# AA1
AA1_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA1") %>% 
  select(query)

AA1_sub <- AA1_hits[duplicated(AA1_hits), ]

# AA1_1
AA1_1_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA1_1") %>% 
  select(query)

AA1_1_sub <- AA1_1_hits[duplicated(AA1_1_hits), ]

# AA1_2
AA1_2_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA1_2") %>% 
  select(query)

AA1_2_sub <- AA1_2_hits[duplicated(AA1_2_hits), ]

# AA2
AA2_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA2") %>% 
  select(query)

AA2_sub <- AA2_hits[duplicated(AA2_hits), ]

# AA3
AA3_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA3") %>% 
  select(query)

AA3_sub <- AA3_hits[duplicated(AA3_hits), ]

# AA3_2
AA3_2_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA3_2") %>% 
  select(query)

AA3_2_sub <- AA3_2_hits[duplicated(AA3_2_hits), ]

# AA4
AA4_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA4") %>% 
  select(query)

AA4_sub <- AA4_hits[duplicated(AA4_hits), ]

# AA5
AA5_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA5") %>% 
  select(query)

AA5_sub <- AA5_hits[duplicated(AA5_hits), ]

# AA5_1
AA5_1_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA5_1") %>% 
  select(query)

AA5_1_sub <- AA5_1_hits[duplicated(AA5_1_hits), ]

# AA6
AA6_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA6") %>% 
  select(query)

AA6_sub <- AA6_hits[duplicated(AA6_hits), ]

# AA10
AA10_hits <- dmnd_out_CAZy %>% 
  select(c(query, classI)) %>% 
  filter(classI == "AA10") %>% 
  select(query)

AA10_sub <- AA10_hits[duplicated(AA10_hits), ]

## export the files to use in linux (these will need to be transferred with FileZilla)
write_csv(ppk_sub, "data/RNA/ppk_kaiju.csv")
write_csv(nap_sub, "data/RNA/nap_sub_kaiju.csv")
write_csv(nar_sub, "data/RNA/nar_sub_kaiju.csv")
write_csv(nirk_sub, "data/RNA/nirk_sub_kaiju.csv")
write_csv(qnor_sub, "data/RNA/qnor_sub_kaiju.csv")
write_csv(AA0_sub, "data/RNA/AA0_sub_kaiju.csv")
write_csv(AA1_sub, "data/RNA/AA1_sub_kaiju.csv")
write_csv(AA1_1_sub, "data/RNA/AA1_1_sub_kaiju.csv")
write_csv(AA1_2_sub, "data/RNA/AA1_2_sub_kaiju.csv")
write_csv(AA2_sub, "data/RNA/AA2_sub_kaiju.csv")
write_csv(AA3_sub, "data/RNA/AA3_sub_kaiju.csv")
write_csv(AA3_2_sub, "data/RNA/AA3_2_sub_kaiju.csv")
write_csv(AA4_sub, "data/RNA/AA4_sub_kaiju.csv")
write_csv(AA5_sub, "data/RNA/AA5_sub_kaiju.csv")
write_csv(AA5_1_sub, "data/RNA/AA5_1_sub_kaiju.csv")
write_csv(AA6_sub, "data/RNA/AA6_sub_kaiju.csv")
write_csv(AA10_sub, "data/RNA/AA10_sub_kaiju.csv")
```

### filterbyname in linux
```{bash}
copy all the fastq.gz samples to the "forKaiju" folder

created .txt files for each of the gene subset files (copy/paste the query column into nano)

# run this to load the package
export PATH=/programs/bbmap-39.03:$PATH
	
# basic function setup
filterbyname.sh in=*.fastq out=*_ppk.fastq names=ppk.txt include=t

# to run one sample at a time:
filterbyname.sh in=100_trim_cut_181049_201_1.fastq out=100_trim_cut_ppk.fastq names=ppk.txt include=t 

# to loop through the samples in each folder for each gene
## (change the "_PN_DRW" as needed for each folder)
for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=ppk.txt include=t overwrite=t &&
    cat $results.fastq.gz >> ppk_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=nap.txt include=t overwrite=t &&
    cat $results.fastq.gz >> nap_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=nar.txt include=t overwrite=t &&
    cat $results.fastq.gz >> nar_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=nirk.txt include=t overwrite=t &&
    cat $results.fastq.gz >> nirk_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=qnor.txt include=t overwrite=t &&
    cat $results.fastq.gz >> qnor_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA0.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA0_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA1.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA1_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA1_1.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA1_1_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA1_2.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA1_2_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA2.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA2_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA3.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA3_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA3_2.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA3_2_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA4.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA4_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA5.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA5_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA5_1.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA5_1_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA6.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA6_PN_DRW.fastq.gz 
done

for f in ./*.fastq;
    do results="$(basename $f .fastq)"
    filterbyname.sh in=$f out=$results.fastq.gz names=AA10.txt include=t overwrite=t &&
    cat $results.fastq.gz >> AA10_PN_DRW.fastq.gz 
done
```

## Kaiju output

### import compiled phyla data from kaiju
```{r}
control_CNS <- read_csv("data/RNA/control_CNS_combined.csv") %>% 
  separate(gene_condition, c("gene", "treatment", "reactor"), sep = "_", remove = FALSE) %>% 
  mutate(percent = (reads/total)*100)

control_DRW <- read_csv("data/RNA/control_DRW_combined.csv") %>%
  separate(gene_condition, c("gene", "treatment", "reactor"), sep = "_", remove = FALSE) %>% 
  mutate(percent = (reads/total)*100)

P_CNS <- read_csv("data/RNA/P_CNS_combined.csv") %>% 
  separate(gene_condition, c("gene", "treatment", "reactor"), sep = "_", remove = FALSE) %>% 
  mutate(percent = (reads/total)*100)

P_DRW <- read_csv("data/RNA/P_DRW_combined.csv") %>% 
  separate(gene_condition, c("gene", "treatment", "reactor"), sep = "_", remove = FALSE) %>% 
  mutate(percent = (reads/total)*100)

PN_CNS <- read_csv("data/RNA/PN_CNS_combined.csv") %>% 
  separate(gene_condition, c("gene", "treatment", "reactor"), sep = "_", remove = FALSE) %>% 
  mutate(percent = (reads/total)*100)

PN_DRW <- read_csv("data/RNA/PN_DRW_combined.csv") %>% 
  separate(gene_condition, c("gene", "treatment", "reactor"), sep = "_", remove = FALSE) %>% 
  mutate(percent = (reads/total)*100)

## Join together
kaiju_compiled <- rbind(control_CNS, control_DRW, P_CNS, P_DRW, PN_CNS, PN_DRW) %>% 
  mutate(treatment = ifelse(treatment == "control", "Control", treatment)) %>%
  mutate(treatment = ifelse(treatment == "PN", "P+N", treatment))
```

### taxonomy plots
```{r}
source("code/colors_and_shapes.R")

## ppk
ppk_taxplot <- kaiju_compiled %>% 
  filter(gene == "ppk") %>% 
  filter(percent > 1) %>%
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(tol(12))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10),
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## nap
nap_taxplot <- kaiju_compiled %>% 
  filter(gene == "nap") %>% 
  filter(percent > 1) %>%
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(tol(12))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## nar
nar_taxplot <- kaiju_compiled %>% 
  filter(gene == "nar") %>% 
  filter(percent > 1) %>%
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(tol(12))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## nirk
nirk_taxplot <- kaiju_compiled %>% 
  filter(gene == "nirk") %>% 
  filter(percent > 1) %>%
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(tol(12))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## qnor
qnor_taxplot <- kaiju_compiled %>% 
  filter(gene == "qnor") %>% 
  filter(percent > 1) %>%
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(tol(12))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA0
AA0_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA0") %>% 
  filter(percent > 1) %>%
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(15))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA1
AA1_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA1") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA1_1
AA1.1_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA1.1") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))
## AA1_2
AA1.2_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA1.2") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA2
AA2_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA2") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(22))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA3
AA3_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA3") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA3_2
AA3.2_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA3.2") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA4
AA4_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA4") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA5
AA5_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA5") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA5_1
AA5.1_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA5.1") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA6
AA6_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA6") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

## AA10
AA10_taxplot <- kaiju_compiled %>% 
  filter(gene == "AA10") %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = phylum_colors) +
  #scale_fill_manual(values=as.vector(stepped(18))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  #theme_classic() +
  theme(axis.text = element_text(size = 10), 
        legend.position = "none",
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

# reorder the data so that most abundant is at the bottom of the graph
ppk_taxplot$data$taxon_name <- reorder(ppk_taxplot$data$taxon_name, ppk_taxplot$data$percent)
nap_taxplot$data$taxon_name <- reorder(nap_taxplot$data$taxon_name, nap_taxplot$data$percent)
nar_taxplot$data$taxon_name <- reorder(nar_taxplot$data$taxon_name, nar_taxplot$data$percent)
nirk_taxplot$data$taxon_name <- reorder(nirk_taxplot$data$taxon_name, nirk_taxplot$data$percent)
qnor_taxplot$data$taxon_name <- reorder(qnor_taxplot$data$taxon_name, qnor_taxplot$data$percent)
AA0_taxplot$data$taxon_name <- reorder(AA0_taxplot$data$taxon_name, AA0_taxplot$data$percent)
AA1_taxplot$data$taxon_name <- reorder(AA1_taxplot$data$taxon_name, AA1_taxplot$data$percent)
AA1.1_taxplot$data$taxon_name <- reorder(AA1.1_taxplot$data$taxon_name, AA1.1_taxplot$data$percent)
AA1.2_taxplot$data$taxon_name <- reorder(AA1.2_taxplot$data$taxon_name, AA1.2_taxplot$data$percent)
AA2_taxplot$data$taxon_name <- reorder(AA2_taxplot$data$taxon_name, AA2_taxplot$data$percent)
AA3_taxplot$data$taxon_name <- reorder(AA3_taxplot$data$taxon_name, AA3_taxplot$data$percent)
AA3.2_taxplot$data$taxon_name <- reorder(AA3.2_taxplot$data$taxon_name, AA3.2_taxplot$data$percent)
AA4_taxplot$data$taxon_name <- reorder(AA4_taxplot$data$taxon_name, AA4_taxplot$data$percent)
AA5_taxplot$data$taxon_name <- reorder(AA5_taxplot$data$taxon_name, AA5_taxplot$data$percent)
AA5.1_taxplot$data$taxon_name <- reorder(AA5.1_taxplot$data$taxon_name, AA5.1_taxplot$data$percent)
AA6_taxplot$data$taxon_name <- reorder(AA6_taxplot$data$taxon_name, AA6_taxplot$data$percent)
AA10_taxplot$data$taxon_name <- reorder(AA10_taxplot$data$taxon_name, AA10_taxplot$data$percent)

#ggarrange(ppk_taxplot, nap_taxplot, nar_taxplot, nirk_taxplot, qnor_taxplot, ncol = 3, nrow = 2)

#ggarrange(AA0_taxplot, AA1_taxplot, AA1.1_taxplot, AA1.2_taxplot, 
          #AA2_taxplot, AA3_taxplot, AA3.2_taxplot, AA4_taxplot, 
          #AA5_taxplot, AA5.1_taxplot, AA6_taxplot, AA10_taxplot, ncol = 4, nrow = 3)

# cutting out AA10 since it only pops up in three groups (P DRW, and both P+Ns)
# and unclassified/unassigned makes up 79% of those on average
# meaning it's not biologically relevant/meaningful
ggarrange(nap_taxplot, nar_taxplot, nirk_taxplot, qnor_taxplot,
          AA0_taxplot, AA1_taxplot, AA1.1_taxplot, AA1.2_taxplot, 
          AA2_taxplot, AA3_taxplot, AA3.2_taxplot, AA4_taxplot, 
          AA5_taxplot, AA5.1_taxplot, AA6_taxplot, ppk_taxplot, ncol = 4, nrow = 4,
          labels = c("nap", "nar", "nirK", "qNor", "AA0", "AA1", "AA1.1", 
                     "AA1.2", "AA2", "AA3", "AA3.2", "AA4", "AA5", "AA5.1", "AA6", "ppk"),
          common.legend = FALSE)

ggarrange(nap_taxplot, nar_taxplot, nirk_taxplot, qnor_taxplot,
          AA0_taxplot, AA1_taxplot, AA1.1_taxplot, AA1.2_taxplot, 
          AA2_taxplot, AA3_taxplot, AA3.2_taxplot, AA4_taxplot, 
          AA5_taxplot, AA5.1_taxplot, AA6_taxplot, ppk_taxplot, ncol = 3, nrow = 6,
          labels = c("nap", "nar", "nirK", "qNor", "AA0", "AA1", "AA1.1", 
                     "AA1.2", "AA2", "AA3", "AA3.2", "AA4", "AA5", "AA5.1", "AA6", "ppk"),
          common.legend = FALSE)

par(mar = c(1, 1, 1, 1))
plot(NULL, xaxt='n', yaxt='n', ylab='', xlab='', bty='n', xlim=c(0,1), ylim=c(0,1))
legend("topleft", ncol = 2, legend =c('Proteobacteria', 'Actinobacteria', 'Unclassified/Unassigned', 'Ascomycota', 'Basidiomycota', 'Firmicutes', 'Bacteroidetes', 'Mucoromycota', 'Chloroflexi', 'Acidobacteria', 'Verrucomicrobia', 'Candidatus Rokubacteria', 'Deinococcus-Thermus', 'Candidate Division NC10', 'Euryarchaeota', 'Planctomycetes', 'Candidatus Handelsmanbacteria', 'Chlorophyta', 'Chytridiomycota', 'Ciliophora', 'Euglenozoa', 'Foraminifera', 'Perkinsozoa', 'Apicomplexa', 'Haptophyta', 'Heterolobosea', 'Microsporidia', 'Bacillariophyta', 'Nitrospirae', 'Oomycota', 'Parabasalia', 'Evosea', 'Rhodophyta', 'Zoopagomycota', 'Cyanobacteria', 'Cercozoa', 'Fornicata', 'Imbricatea', 'Viruses'), pch=15, pt.cex=3, cex=1.2, bty='n',
    col = c('gray20', '#0F6B99FF', '#7EC3E5FF', '#B2E5FFFF', '#6B990FFF', '#A3CC51FF', '#E5FFB2FF', '#99540FFF', '#CC8E51FF', '#FFD8B2FF', '#990F0FFF','#CC5151FF', '#FFB2B2FF', '#422CB2FF', '#8F7EE5FF', '#BFB2FFFF', 'grey', 'gray50', 'royalblue', 'darkorange', 'olivedrab', 'greenyellow', 'green', 'darkorchid1', '#657CDC', '#A77D87', '#55AD91', '#D47FE7', '#D7E63E', 'goldenrod', '#9BA195', '#B0A2E8', '#92AB76', '#DBB46E', '#E355A9', '#A8E8EA', '#E3C2DD', '#E38B80', '#5ABF66'))
mtext("Phylum", at=0.1, cex=1.2)
```


## Taxonomy for CNS vs DRW
```{r}
CNS_phyla <- read_csv("data/RNA/CNS_phyla_kaiju.csv")
DRW_phyla <- read_csv("data/RNA/DRW_phyla_kaiju.csv")

phyla_combined <- rbind(CNS_phyla, DRW_phyla)

phyla_combined %>% 
  mutate(percent = (reads/total)*100) %>% 
  filter(percent > 1) %>% 
  ggplot(aes(x = reactor, y = percent, fill = fct_reorder(taxon_name, percent))) +
  geom_bar(stat = "identity", position = "fill") +
  guides(fill = guide_legend(reverse=TRUE)) +
  facet_grid(timepoint~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values=rev(as.vector(stepped2(20)))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  theme_classic() +
  theme(axis.text = element_text(size = 12), 
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))

phyla_combined %>% 
  mutate(percent = (reads/total)*100) %>% 
  filter(percent > 2) %>% 
  ggplot(aes(x = reactor, y = percent, fill = taxon_name)) +
  geom_bar(stat = "identity", position = "fill") +
  #facet_grid(timepoint~treatment) +
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values=as.vector(stepped(16))) +
  xlab("Reactor") +
  ylab("Relative Abundance") +
  labs(fill = "Phylum") +
  theme_classic() +
  theme(axis.text = element_text(size = 12), 
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))
```
