---
title: "RNA Analysis - Chapter 2 work"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Files needed:

- 

## Load packages & set directory
```{r, include=FALSE}
#install.packages("tidyverse")
library(tidyverse)

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("dada2", version = "3.18")
library(dada2)
#packageVersion("dada2")
```

## set path to seq files
```{r}
# Set path to the gzipped files
path <- "E:/RNA_kld"
path

# What files do we have?
list.files(path)

# Setting a variable with all the sample names by scanning our "samples.txt" file
samples <- scan(file = "data/samples.txt", what = "character")
samples
```

## Load forward and reverse reads
```{r}
# Create a variable for the forward and reverse reads

#1. Forward read variable
forward_reads <- sort(list.files("E:/RNA_kld", pattern = "_R1.fastq.gz", full.names = TRUE))
forward_reads

#2. Reverse read variable
reverse_reads <- sort(list.files("E:/RNA_kld", pattern = "_R2.fastq.gz", full.names = TRUE))
reverse_reads

# Show the quality of each base on the reads of first 4 samples
fwdQual4_plot <- plotQualityProfile(forward_reads[1:4])
revQual4_plot <- plotQualityProfile(reverse_reads[1:4])

#3. Place trimmed/filtered files into /trimmed_filtered/ subdirectory
## Create a variable holding the files names for the forward and reverse reads
filtered_fwd_reads <- file.path(path, "trimmed_filtered", paste0(samples, "_R1_trimmed_filtered.fastq"))
filtered_fwd_reads

filtered_rev_reads <- file.path(path, "trimmed_filtered", paste0(samples, "_R2_trimmed_filtered.fastq.gz"))
filtered_rev_reads
```

#### Filter and trim for quality (trimLeft)
```{r filter-trim}
#FWD_adapter <- "AGATCGGAAGAGC"  # CHANGE ME to your forward primer sequence
#REV_primer <- "xyz"  # CHANGE ME to your reverse primer sequence
FWD_primer_length <- 15 # CHANGE ME to your forward primer sequence length
REV_primer_length <- 15 # CHANGE ME to your reverse primer sequence length

# Trimming cut-offs were selected via visual inspection of the quality plots
filtered_out <- filterAndTrim(fwd = forward_reads, 
                              filt = basename(filtered_fwd_reads), 
                              rev = reverse_reads, 
                              filt.rev = basename(filtered_rev_reads),
                              trimLeft = c(FWD_primer_length, REV_primer_length), 
                              maxN = 0,
                              multithread = FALSE)
# maxN: every base counts with ASV, so we don't want any ambiguous sequences
# Got error: "multithreading has been DISABLED, as forking is not supported on .Platform$OS.type 'windows'"
## Set multithreading = FALSE on Windows (takes less time and R will disable it anyway)
## Zymo Q&A states this is not a priority error so I left it alone

# Plot the quality of trimmed reads
trimfilt_fwdQual4_plot <- plotQualityProfile(filtered_fwd_reads[1:4])
trimfilt_revQual4_plot <- plotQualityProfile(filtered_rev_reads[1:4])
```

### Percent Reads Lost (compare methods)
```{r}
head(filtered_out)

# Remove the file type from the sample row names to make the names match for joining
for (row in 1:nrow(filtered_out)){
  rownames(filtered_out)[row] <- sub("_R1_trimmed.fastq.gz", "-R1", rownames(filtered_out)[row])
}

# Revise the column names and format for joining
percent_loss <- filtered_out %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "sample") %>% 
  as_tibble() %>%
  mutate(cutadapt = ((reads.in - reads.out)/reads.in*100)) %>% 
  select(sample, cutadapt)

# Bar plot of reads in vs. reads out
percent_loss_barplot <- percent_loss %>% 
  pivot_longer(cols = c(cutadapt, trimLeft), names_to = "method", values_to = "percent_loss") %>%   
  ggplot(aes(x = sample, y = percent_loss, fill = method)) +
  geom_bar(position = "dodge", stat = "identity") +
  theme(axis.text.x = element_text(angle = 90), legend.position = "top", legend.title = element_blank()) +
  ylim(0,100) +
  labs(title = "Percent loss of reads via different methods of trimming/filtering") +
  xlab("Sample ID") +
  ylab("% Loss")
percent_loss_barplot

# Mean sequence length (post trim)
filtered_out_trimLeft %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "sample") %>% 
  as_tibble() %>% 
  summary(reads.out)
```
