---
title: "Taxonomy - data from KBase Kaiju"
author: "Katie L. Duggan"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Files needed:

- "kaiju_taxonomy.csv" (output from Kaiju app in KBase - open .kaiju file in notepad and pasted data into csv file)

## Load packages
```{r, include=FALSE}
library(tidyverse)
#install.packages("pals")
library(pals)
library(vegan)
library(glue)
library(broom)
library(ggrepel)
library(rstatix)
```

## Load data
```{r}
taxonomy <- read_csv("C:/github/WBR-microbiome/data/kaiju_output_100nounclassified.csv")
### "kaiji_summaries.zip" was downloaded from KBase Kaiju app
### Phylum data was pasted into csv file
### Unclassified data is excluded

### "Unclassified" data was used to calculate total reads, then phylum percentages were re-calculated based on that total
```

## Taxonomy plot
```{r}
# Sort out low abundance phyla
taxa_nolowabund <- taxonomy %>% 
  dplyr::filter(percent > 1) %>% # Remove phyla that are < 1% abundant
  dplyr::filter(!taxon_name == "cannot be assigned to a (non-viral) phylum") %>% 
  dplyr::filter(!taxon_name == "belong to a (non-viral) phylum with less than 0.5% of all reads") %>%
  mutate(taxon_name = replace(taxon_name, taxon_name == 'Ascomycota', 'Ascomycota (fungi)')) %>% 
  mutate(name = ifelse(name == "7G", "M1", name)) %>% 
  mutate(name = ifelse(name == "9G", "M2", name)) %>% 
  mutate(name = ifelse(name == "11G", "M3", name)) %>% 
  mutate(name = ifelse(name == "8W", "U1", name)) %>% 
  mutate(name = ifelse(name == "10W", "U2", name)) %>% 
  mutate(name = ifelse(name == "12W", "U3", name))

sample_order <- c('M1', 'M2', 'M3', 'U1', 'U2', 'U3') 

phyla_order <- c('Proteobacteria', 'Ascomycota (fungi)', 'Bacteroidetes', 'Chloroflexi', 'Verrucomicrobia', 'cannot be assigned to a (non-viral) phylum', 'belong to a (non-viral) phylum with less than 0.5% of all reads', 'Acidobacteria', 'Planctomycetes', 'Actinobacteria', 'Ignavibacteriae', 'Firmicutes')

phyla_order_reverse <- rev(phyla_order)

taxa_nolowabund %>%
  ggplot(aes(x = factor(name, level = sample_order), y = percent, fill = factor(taxon_name, level = phyla_order_reverse))) +
  #fill = fct_reorder(taxon_name, percent))) +
  geom_bar(stat = "identity", position = "fill") +
  facet_grid(~treatment, scales = "free") +
  scale_y_continuous(name = "Taxa Relative Abundance (%)", labels = scales::percent) +
  scale_fill_manual(values=as.vector(tol(12))) +
  xlab("Sample ID") +
  labs(fill = "Phylum") +
  theme_classic() +
  theme(axis.text = element_text(size = 12), 
        axis.title = element_text(size = 12), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 12), 
        strip.text = element_text(size = 12))
```


## Statistics
Subset the data by treatment:
```{r}
ascomycota <- taxa_nolowabund %>% 
  filter(taxon_name == "Ascomycota (fungi)")
```

t test for read count of fungi in U vs M samples:
```{r}
t_test(reads ~ treatment, data = ascomycota, alternative = "two.sided", paired = FALSE, conf.level = 0.95)
```


## Clustering exercise
```{r}
taxa_nolowabund

taxa_for_pcoa <- taxa_nolowabund %>% 
  mutate(total_reads = (reads*100)/percent)

### Follow Riffomonas tutorials from here:
### https://www.youtube.com/watch?v=xyufizOpc5I
### https://www.youtube.com/watch?v=G5Qckqq5Erw
### https://www.youtube.com/watch?v=_EjNR6YdnGM
```

#### Get matrix form
```{r}
shared <- taxa_for_pcoa %>% 
  select(name, reads, taxon_name) %>% 
  rename("value" = "reads") %>%
  select(name, taxon_name, value) %>%
  rename("Group" = "name") %>% 
  rename("name" = "taxon_name") %>% 
  group_by(Group) %>% 
  #summarize(total = sum(value)) %>% 
  #arrange(total) #this shows that 10W is the lowest at 15700522 reads
  mutate(total = sum(value)) %>%
  #filter(total > 15700521) %>% 
  group_by(name) %>% 
  mutate(total = sum(value)) %>% 
  filter(total != 0) %>% 
  ungroup() %>% 
  select(-total) %>% 
  pivot_wider(names_from = name, values_from = value) %>% 
  replace(is.na(.), 0) %>%
  as.data.frame()

rownames(shared) <- shared$Group
shared <- shared[, -1] #remove name column
shared <- as.matrix(shared) #input for the functions from vegan (rows are samples, columns are taxa, values are the counts)
```

#### Calculating distances 
```{r}
set.seed(8675309)
dist <- avgdist(shared, dmethod = "bray", sample = 180000) #rarefied based on lowest sample total (10W = 15700522)
nmds <- metaMDS(dist)

scores(nmds) %>% 
  as_tibble(rownames = "Group") %>%
  mutate(treatment = c("Milled", "Unmilled", "Milled", "Unmilled", "Milled", "Unmilled")) %>% 
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point()
```

#### Build the ordination
```{r}
dist_matrix <- dist #to match riffomonas terminology

pcoa_tax <- cmdscale(dist_matrix, eig = TRUE, add = TRUE)
pcoa_tax

positions <- pcoa_tax$points
positions

colnames(positions) <- c("pcoa1", "pcoa2")

percent_explained <- 100* pcoa_tax$eig / sum(pcoa_tax$eig)

pretty_pe <- format(round(percent_explained[1:2], digits = 1), nsmall = 1, trim = TRUE)

labs <- c(glue("PCo 1 ({pretty_pe[1]}%)"),
          glue("PCo 2 ({pretty_pe[2]}%)"))

## Main formatted plot for paper:
positions %>% 
  as_tibble(rownames = "samples") %>% 
  mutate(treatment = c("Milled", "Unmilled", "Milled", "Unmilled", "Milled", "Unmilled")) %>% 
  ggplot(aes(x = pcoa1, y = pcoa2, color = treatment)) +
  geom_point(size = 3) #+
  #labs(x = labs[1], y = labs[2], color = "Treatment")

tibble(pe = percent_explained,
       axis = 1:length(percent_explained)) %>% 
  ggplot(aes(x = axis, y = pe)) +
  geom_line() #+
  #coord_cartesian(xlim = c(1,10))

cumsum(percent_explained) # you'd have to extend to the 4th axis to get 100% of the data explained
```

#### PERMANOVA
```{r}
metadata <- read.csv("C:/github/WBR-microbiome/data/pcoa_metadata.csv")

adonis2(dist_matrix ~ treatment, data = metadata)
#p-value = 0.1
```
